src/ensembl/src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects.tsx

```diff
@@ -0,0 +1,207 @@
+                    description={'Remove'}
+                    image={RemoveIcon}
+                  />
+                </div>
+              )}
+            </td>
+          </tr>
+
+          {selectedOptionsClone.map((selectedOption: any, key: number) => {
+            const selectOptions = [...props.selectOptions]
+              .filter((option: any) => {
+                if (
+                  (selectedOptionsClone.indexOf(option.value) == -1 &&
+                    firstSelectedOption !== option.value) ||
+                  selectedOption === option.value
+                ) {
+                  return true;
+                }
+                return false;
+              })
+              .map((option: any) => {
+                const optionClone = { ...option };
+                if (optionClone.value === selectedOption) {
+                  optionClone.isSelected = true;
+                }
+                return optionClone;
+              });
+
+            return (
+              <tr key={key}>
+                <td />
+                <td>
+                  <div>
+                    <Select
+                      options={selectOptions}
+                      onSelect={(option: string) => {
+                        handleOnSelect(option, key + 1);
+                      }}
+                      placeholder={'select'}
+                    />
+                  </div>
+                </td>
+                <td className={styles.removeIcon}>
+                  <div className={styles.removeIconHolder}>
+                    <ImageButton
+                      onClick={() => removeSelection(selectedOption)}
+                      description={'Remove'}
+                      image={RemoveIcon}
+                    />
+                  </div>
+                </td>
+              </tr>
+            );
+          })}
+
+          {shouldShowExtraOption && (
+            <tr>
+              <td />
+              <td>
+                <div>
+                  <Select
+                    options={newSelectOptions}
+                    onSelect={(option: string) => {
+                      handleOnSelect(option);
+                    }}
+                    placeholder={'Select'}
+                  />
+                </div>
+              </td>
+              <td />
+            </tr>
+          )}
+
+          {/* Show the Add button */}
+          {props.selectedOptions.length > 0 &&
+            !shouldShowExtraOption &&
+            newSelectOptions.length > 0 && (
+              <tr>
+                <td colSpan={2} />
+                <td>
+                  <div className={styles.addIconHolder}>
+                    <ImageButton
+                      onClick={() => setShowExtraOption(true)}
+                      description={'Add'}
+                      image={AddIcon}
+                    />
+                  </div>
+                </td>
+              </tr>
+            )}
+        </tbody>
+      </table>
+    </>
```

Comment:
- no need for fragment
- does it need to be a table?

Reply:
Removed the fragment.

I feel it is easier to control the vertical alignment of the Select, add Icon & remove icon, by using a table.